### PR TITLE
dev: template models, initdb, CRUD APIs

### PR DESCRIPTION
**Pull request by PeterBaker0**
Implements the CRUD APIs to interact with new template DB. Also adds new API route to create a new notebook from an existing template.

Follows the threads around app to initialise the DB, fixes security model issue where members list was empty. 

**Note**: Uses existing `cluster-admin` role for now - we can consider if fine-grained permissions are needed and can add this in using the combination of init methods, security documents, and the API user management (which defines actions against user roles etc). 

Fixes/improves various aspects along the way including using Zod to do runtime validation of payloads, fully typing the templates DB so that PouchDB methods are type safe, replacing default HTML error handler with JSON handler and implementing custom exceptions so that status codes automatically propagate to the error middleware.

### CODE REVIEW

**Review by stevecassidy — CHANGES_REQUESTED** [review]
Main request is for some tests, I'm not able to try this code out without them.

File: library/data-model/src/types.ts [hunk 1]
```diff
@@ -678,3 +680,103 @@ export interface InitialMergeDetails {
   initial_head: RevisionID;
   initial_head_data: RecordMergeInformation;
 }
+
+// ================== MOCKS FOR NEW SURVEY API ===============
+// Template database item
+// These are the fields used to instantiate the item
+export const UiSpecificationSchema = z.record(z.any());
+export const NotebookMetadataSchema = z.record(z.string());
+
```
[the remark this answers is not in the document]
Comment (PeterBaker0) [reply library/data-model/src/types.ts:714]: @stevecassidy can we defer doing this - could be a pretty big job and out of scope for this set of features. 

https://jira.csiro.au/browse/BSS-340

[the remark this answers is not in the document]
Comment (stevecassidy) [reply library/data-model/src/types.ts:714]: Sure

File: api/src/authkeys/read.ts [hunk 2]
```diff
@@ -28,14 +28,15 @@ import {KEY_SERVICE} from '../buildconfig';
  * @returns Details of the verified user or undefined
  */
 export const validateToken = async (token: string) => {
+  const debug = false;
   const signingKey = await KEY_SERVICE.getSigningKey();
```
Comment (stevecassidy) [inline api/src/authkeys/read.ts:32]: Maybe just remove these, a local flag isn't useful.

**Review by stevecassidy — APPROVED** [review]
New changes look good. Really nice work on the tests.  One minor change to remove logging output.

**Comment by PeterBaker0** [comment]
Merging as changes were as discussed, and unit tests are passing.

**Pull request merged**